ui/lib/components/ColumnsSelector/index.tsx

```diff
@@ -119,19 +119,28 @@ export default function ColumnsSelector({
   )
 
   const content = (
-    <Space direction="vertical">
-      {filteredColumns.map((column) => (
-        <Checkbox
-          key={column.key}
-          checked={visibleKeys[column.key]}
-          onChange={(e) => handleCheckChange(e, column)}
-        >
-          {column.name}
-        </Checkbox>
-      ))}
-
+    <div style={{ marginTop: -12 }}>
+      <Space
+        direction="vertical"
+        style={{
+          maxHeight: 400,
+          overflow: 'auto',
+          paddingTop: 8,
+          paddingBottom: 8,
+        }}
+      >
+        {filteredColumns.map((column) => (
```

Comment:
There are too many items here. We need to provide some filters. Maybe we can use `MultiSelect`?

Reply:
This PR is already too huge, how about leave it to the next PR for refining?